qiime2/sdk/usage.py

```diff
@@ -1210,6 +1210,123 @@ def factory():
             return artifact
         return self._usage_variable(name, factory, 'artifact')
 
+    def construct_collection(
+        self, name: str, member_type: str, members: Union[dict, list]
+    ) -> UsageVariable:
+        '''
+        Return a UsageVariable of type artifact_collection given a list or dict
+        of its members.
+
+        Parameters
+        ----------
+        name : str
+            The name of the resulting variable.
+        member_type : str
+            One of 'artifact', 'visualization'. The types of the members in the
+            collection.
+        members: list or dict
+            The desired members of the ResultCollection.
+
+        Returns
+        -------
+        UsageVariale
+            Of type artifact_collection or viz_collection.
+
+        Examples
+        --------
+        >>> mapping_1, = use.action(
+        ...     use.UsageAction('dummy_plugin', 'params_only_method'),
+        ...     use.UsageInputs(name='c', age=100),
+        ...     use.UsageOutputNames(out='mapping_1')
+        ... )
+        >>> mapping_1
+        <ExecutionUsageVariable name='mapping_1', var_type='artifact'>
+        >>> collection_1 = use.construct_collection(
+        ...     'collection_1', 'artifact', {'a': mapping_1, 'b': mapping_1}
+        ... )
+        >>> collection_1
+        <ExecutionUsageVariable name='collection_1', var_type='artifact_collection'>
+        '''  # noqa: E501
+        if member_type not in ('artifact', 'visualization'):
+            msg = (
+                'Collections must be either of type artifact or of type '
+                f'visualization, not {member_type}.'
+            )
+            raise ValueError(msg)
+
+        if not all(
+            member.var_type == member_type for member in members.values()
+        ):
+            msg = (
+                'Expected either only artifacts or only visualizations in a '
+                'collection.'
+            )
+            raise ValueError(msg)
+
+        def factory():
+            from qiime2 import ResultCollection
+            # NOTE: these usage variables are assumed to have been
+            # materialized at this point
+            members_dict = {
+                key: member.value for key, member in members.items()
+            }
+
+            return ResultCollection(members_dict)
+
+        if member_type == 'artifact':
+            return self._usage_variable(name, factory, 'artifact_collection')
+
+        return self._usage_variable(name, factory, 'viz_collection')
+
+    def access_collection_member(
```

Comment:
I think this should probably be `get_collection_member` to match `get_metadata_column` which I think also implies that the `collection` parameter should be `variable` to also match.